### ISSUE TITLE
going over the queryLimit result has a breaking change in v5. But should it be allowed at all?

### ISSUE THREAD

**Issue by laurenskling** [strapi/strapi#22483]
Sorry for not following the template, but this is more of a question than a report.

In v4, when I set the rest maxLimit to 100, and I query with pageSize=250 (which is currently the `gatsby-source-strapi` default limit), the result would be 100 entries _from (100 * `page`)_. So 1371 will have 14 pages, and `[pageSize]=250&pagination[page]=13` will actually hold items 1301-1371

In v5, when I set the rest maxLimit to 100, and I query with pageSize=250 (which is currently the `gatsby-source-strapi` default limit), the result would be 100 entries _from (250 * `page`)_. After the 7th page, `pagination[pageSize]=250&pagination[page]=7`, there will be zero items, as it starts getting item 1500. 
To make things more interesting, the response `meta.pagination.page` actually sees i'm going over the limit, as it returns 16 (we were requesting page 7):
```
{
    "data": [],
    "meta": {
        "pagination": {
            "page": 16,
            "pageSize": 100,
            "pageCount": 14,
            "total": 1367
        }
    }
}
```
    
This raises the question, should the REST call not crash on going over the limit? Because the response will not be what I am asking, so I'd rather have no response than a bad response. I just learned that the hard way the past 2 days...

**tl;dr
going over the rest maxLimit is flaky. Can we please make it crash?**

**Comment by Boegie19** [comment]
My main question I am asking my self is if we fix this in V5 would that not be an other breaking change.
since this breaking change is allowed since V4 to V5 but if we do one now mid V5 we mind break projects who are made with V5 who expect us not to error.

**Comment by laurenskling** [comment]
I understand where you are coming from but I dont agree.
In v4, you would actually get a correct result set. Pagination would be correct for the maxLimit, no matter what you did.
in v5, you will get max results, but a pagination using the limit you asked for. Which is just incorrect. A bug. I would concider it a feature to help spotting errors. Not a breaking change.

**Comment by derrickmehaffy** [comment]
Let me ask the engineers their thoughts as honestly I'm on the fence if it should throw a validation error or not :thinking:

**Comment by Boegie19** [comment]
@derrickmehaffy Still lets say it should be a validation error would chanching it now not be a breaking change?

**Comment by derrickmehaffy** [comment]
I don't know that's why I'm asking the engineering team for their input.

**Comment by Convly** [comment]
From what I've understood so far, the `maxLimit` is basically not respected when returning the documents.

What we could do first is make the v5 behavior isomorphic with v4 (fixing the BC) and then think about where we want to go.
Throwing would indeed be a breaking change in how the API responds

**Comment by laurenskling** [comment]
Nice. Just started working on a large database again and was missing data. Felt suspiciously familiar. 🥹

**Comment by laurenskling** [comment]
Still think this is a bug, not a feature.

**Comment by laurenskling** [comment]
Still think this is a bug, not a feature.
